recipes/ray-packages/meta.yaml

```diff
@@ -0,0 +1,256 @@
+        # autoscaler does not work on Windows, no point of building it
+        - {{ pin_subpackage('ray-autoscaler', exact=True) }}  # [not win]
+    test:
+      imports:
+        # dummy test; actual tests are in subpackages
+        - ray
+
+  - name: ray-core
+    script: build-core.sh  # [not win]
+    script: build-core.bat  # [win]
+    requirements:
+      build:
+        - {{ compiler('c') }}
+        - {{ compiler('cxx') }}
+        - cython >=0.29
+        - bazel <=3.4.1  # [not win]
+        - bazel  # [win]
+        - curl
+        - make
+      host:
+        # TODO: check which of these can move to run-reqs
+        - click
+        - cloudpickle
+        - colorama
+        - filelock
+        - funcsigs
+        - jsonschema
+        - numpy >=1.16
+        - pickle5  # [py<38]
+        - packaging
+        - protobuf >=3.8.0
+        - pytest
+        - pyyaml
+        - redis-py >=3.3.2, <3.5.0
+        - python
+        - pip
+      run:
+        - python
+        - aiohttp
+        - aiohttp-cors
+        - aioredis
+        - click >=7.0
+        - colorama
+        - colorful
+        - filelock
+        - googlesearch
+        # gpustat-0.6.0 has a dependency which does not exist on Windows;
+        # skip it there until gpustat is fixed as it is optional
+        - gpustat  # [not win]
+        - grpcio >=1.28.1
+        - jsonschema
+        - msgpack-python >=1.0.0, <2.0.0
+        - numpy >=1.16
+        - protobuf >=3.8.0
+        - psutil
+        - pyyaml
+        - requests
+        - redis-py >=3.3.2, <3.5.0
+        - opencensus
+        - prometheus_client >=0.7.1
+        - pickle5  # [py<38]
+        - funcsigs
+        - setproctitle
+    test:
+      imports:
+        - ray
+        - ray._raylet
+        - ray.actor
+        - ray.profiling
+        - ray.runtime_context
+        - ray.state
+        - ray.worker
+
+  - name: ray-debug
+    build:
+      # some weird incompatibility when installing during tests... disabling for now
+      skip: True  # [not win]
+      # there is no py-spy on Windows
+      skip: True  # [win]
+    requirements:
+      host:
+        - python
+      run:
+        - python
+        - {{ pin_subpackage('ray-core', exact=True) }}
+        - py-spy >=0.2.0
+    test:
+      imports:
+        # there doesn't appear to be a debug specific module
+        - ray
+
+  - name: ray-dashboard
+    requirements:
+      host:
+        - python
+      run:
+        - python
+        - {{ pin_subpackage('ray-core', exact=True) }}
+    test:
+      imports:
+        - ray.dashboard
+
+  - name: ray-rllib
+    build:
+      # needed atari deps are missing on Windows
+      skip: True  # [win]
+      entry_points:
+        - rllib = ray.rllib.scripts:cli
+    requirements:
+      host:
+        - python
+      run:
+        - python
+        - {{ pin_subpackage('ray-core', exact=True) }}
+        - tabulate
+        - tensorboardX
+        - pandas
+        - dataclasses  # [py < 37]
```

Comment:
I think the preferred style is  `# [py<37]` (happens a few more times in the recipe as well)

Reply:
I find no-space conditions harder to read personally, what is the point of removing the spaces?..

Reply:
We literally only use `# [py<37]` in all other recipes. We don't have an explicit style guide but in keeping it consistent with the rest of the recipes, we should stick to no-spaces here.

Reply:
Okay, fixed in 624613c